sdk/cosmos/azure-cosmos/azure/cosmos/container.py

```diff
@@ -677,6 +677,40 @@ def patch_item(
             response_hook(self.client_connection.last_response_headers, result)
         return result
 
+    @distributed_trace
+    def execute_item_batch(
+            self,
+            batch_operations: List[Tuple[Any]],
+            partition_key: Union[str, int, float, bool],
+            **kwargs) -> Dict[str, Any]:
+        """ Executes the transactional batch for the specified partition key.
+
+        :param batch_operations: The batch of operations to be executed.
+        :type batch_operations: List[Dict[str, Any]]
+        :param partition_key: The partition key value of the batch operations.
+        :type partition_key: Union[str, int, float, bool]
+        :keyword str pre_trigger_include: trigger id to be used as pre operation trigger.
+        :keyword str post_trigger_include: trigger id to be used as post operation trigger.
+        :keyword str session_token: Token for use with Session consistency.
+        :keyword str etag: An ETag value, or the wildcard character (*). Used to check if the resource
```

Comment:
How does etag (and match_condition) work as a param here - doesn't it need to be provided on a per-item basis?

Reply:
it can be provided on per-item basis or also provided for the operation as a whole - just depends on what the user is looking to do with their optimistic concurrency